### PR TITLE
feat: add spacy text encoder

### PR DESCRIPTION
**Pull request by rutujasurve94**
`SpacyTextEncoder` for word (vector) embeddings from text:

-  Uses `tok2vec` component

-  `en_core_web_sm` language

-  Follows `HashEmbedCNN` model

### CODE REVIEW

File: encoders/nlp/SpacyTextEncoder/README.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+  - Specify the image name along with the version tag. The snippet below uses Jina version `1.0.16`
+
+  - ```bash
+    docker pull jinahub/pod.encoder.spacyencoder:0.0.1-1.0.16
+    ```
+   
+ Note:
+ 
+ One of the limitations with the Hub Executors currently is the tags - all Executor images should have the versions appended in the name i.e.
```
Comment (Yongxuanzhang) [inline encoders/nlp/SpacyTextEncoder/README.md:81]: I remember this is fixed by Han, we can use the name of the executor

File: encoders/nlp/SpacyTextEncoder/__init__.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+from jina.executors.decorators import batching, as_ndarray
+from jina.executors.encoders.frameworks import BaseTorchEncoder
+
+class SpacyTextEncoder(BaseTorchEncoder):
+    """
+    :class:`SpacyTextEncoder` encodes data from a `np.ndarray` (of strings) of length `BatchSize` into
+    a `np.ndarray` of shape `Batchsize x EmbeddingDimension`.
+
+    :param lang: pre-trained spaCy language pipeline (model name HashEmbedCNN by default for tok2vec), `en_core_web_sm` by default.
```
Comment (Yongxuanzhang) [inline encoders/nlp/SpacyTextEncoder/__init__.py:14]: Maybe we can add the possible values for 'lang'?

**Review by Yongxuanzhang — COMMENTED** [review]
Overall look good to me. 😄

File: encoders/nlp/SpacyTextEncoder/__init__.py (hunk 2 above)
Comment (rutujasurve94) [reply encoders/nlp/SpacyTextEncoder/__init__.py]: Sure, done.

File: encoders/nlp/SpacyTextEncoder/README.md (hunk 1 above)
Comment (rutujasurve94) [reply encoders/nlp/SpacyTextEncoder/README.md]: Removing this.

**Review by Yongxuanzhang — APPROVED** [review]

**Pull request merged**